src/stream/utils.ts

```diff
@@ -0,0 +1,19 @@
+import * as Etebase from "etebase";
+import { FileSystem } from "react-native-unimodules";
+
+export async function cacheToFile(filename: string, data: string) {
+  const uri = `${FileSystem.cacheDirectory}${encodeURI(filename)}`;
+  await FileSystem.writeAsStringAsync(uri, data);
+  return uri;
+}
+
+export async function getItemData(item: Etebase.Item) {
+  const { name } = item.getMeta();
+  const content = await item.getContent(Etebase.OutputFormat.String);
+  return { name, content };
+}
+
+export function getFilename(name: string, extension = ".md") {
+  const escaped = name.replace(/\//g, "-");
```

Comment:
You probably want to change other characters. Could you maybe looks online for recommended filename sanitization functions? (maybe this one is sufficient, but I think maybe not)

Reply:
You're right, it's not. I wrongly trusted a stackoverflow answer. It's dependent on the filesystem and there are more reserved characters on FAT32. I'll look into it.

Reply:
Don't overdo it, but maybe also backslash at least, and maybe colon as well. We want to be flexible still.

Reply:
I could just use that : https://www.npmjs.com/package/sanitize-filename

Reply:
I think it maybe does too much. E.g. `Ideas? Nah.md` is probably a fine filename for us to have. OK, maybe let's not worry about this for now. Maybe just handle these:

> Control characters (0x00–0x1f and 0x80–0x9f)
> Reserved characters (/, ?, <, >, \, :, *, |, and ")

Reply:
I'm thinking maybe we shouldn't even care about it. I tested on Android and it replaces / with _ for example. So maybe let the OS handle it?

Reply:
Even better!

Reply:
Actually I was wrong, I still need to escape the filename to write a local file first. I was trying with the export feature where it works…

Reply:
https://www.npmjs.com/package/filenamify does just what you asked

Reply:
Cool, just use `-` as the replacement.

Reply:
Yeah, `!` is an insane default

Reply:
Metro shows this warning because of filenamify, but it still works: `Package filenamify has been ignored because it contains invalid configuration. Reason: Package subpath './package.json' is not defined by "exports" in /home/cakeh/Dev/etesync-notes/node_modules/filenamify/package.json`

Reply:
Resolving, we don't need it anymore because we share the content directly